src/test/ui/parser/mut-patterns.stderr

```diff
@@ -1,8 +1,8 @@
-error: expected one of `:`, `;`, `=`, or `@`, found `{`
+error: expected one of `:`, `;`, `=`, `@`, or `|`, found `{`
   --> $DIR/mut-patterns.rs:5:17
    |
 LL |     let mut Foo { x: x } = Foo { x: 3 };
-   |                 ^ expected one of `:`, `;`, `=`, or `@` here
+   |                 ^ expected one of `:`, `;`, `=`, `@`, or `|` here
 
```

Comment:
Nice example of not allowed `mut` patterns. It should have a custom error though.

Reply:
Sure, want me to fix it here or file an issue (since it's mostly unrelated to this PR)?

Reply:
I'll file a ticket. Separate PR worthy.

Reply:
Thanks!